File: javascript/secret-handshake/secret-handshake.js
function zeroFill(number, width) {
  width -= number.toString().length;
  if (width > 0) {
    return number + new Array(width + (/\./.test(number) ? 2 : 1)).join('0');
  }
  return number + ""; // always return a string
}

var SecretHandshake = function (handshake) {
  if (isNaN(handshake)) {
    throw new Error('Handshake must be a number');
  }
  this.handshake = (handshake >>> 0).toString(2);
};

SecretHandshake.prototype.commands = function () {
  var commands = [];
  for (var c = this.handshake.length - 1; c >= 0; c--) {
    var command = zeroFill(+this.handshake[c], this.handshake.length - (c));
    switch (command) {
      case '1':
        commands.push("wink");
        break;
      case '10':
        commands.push("double blink");
        break;
      case '100':
        commands.push("close your eyes");
        break;
      case '1000':
        commands.push("jump");
        break;
      case '10000':
        commands = commands.reverse();
        break;
    }
  }
  return commands;
};

module.exports = SecretHandshake;

File: csharp/secret-handshake/SecretHandshake.cs
using System;
using System.Collections.Generic;

class SecretHandshake
{
  public static string[] Commands(int command)
  {
    var result = Convert.ToString(command, 2);
    var commands = new List<string>();
    for (var c = result.Length - 1; c >= 0; c--)
    {
      var current = char.GetNumericValue(result[c]).ToString().PadRight(result.Length - c, '0');
      switch (current)
      {
        case "1":
          commands.Add("wink");
          break;
        case "10":
          commands.Add("double blink");
          break;
        case "100":
          commands.Add("close your eyes");
          break;
        case "1000":
          commands.Add("jump");
          break;
        case "10000":
          commands.Reverse();
          break;
      }
    }
    return commands.ToArray();
  }
}